connector/src/main/scala/com/datastax/spark/connector/DatasetFunctions.scala

```diff
@@ -54,18 +76,20 @@ class DatasetFunctions[K: Encoder](dataset: Dataset[K]) extends Serializable {
               throw missingColumnException(partitionKeyName, "Partition Key Column")))
           .map(_.copy(columnRole = PartitionKeyColumn))
       ,
-      clusteringColumns = clusteringKeyNames
-          .map(clusteringKeyName =>
-            columnMapping.getOrElse(clusteringKeyName,
-              throw missingColumnException(clusteringKeyName, "Clustering Column")))
+      clusteringColumns = clusteringKeyColumns
+          .map(clusteringKey =>
+            (columnMapping.getOrElse(clusteringKey._1,
+              throw missingColumnException(clusteringKey._1, "Clustering Column")),
+              clusteringKey._2))
           .zipWithIndex
-          .map { case (col, index) => col.copy(columnRole = ClusteringColumn(index))}
+          .map { case (col, index) => col._1.copy(columnRole = ClusteringColumn(index, col._2))}
       ,
       regularColumns = regularColumnNames
           .map(regularColumnName =>
              columnMapping.getOrElse(regularColumnName,
                   throw missingColumnException(regularColumnName, "Regular Column")))
           .map(_.copy(columnRole = RegularColumn))
+      , ifNotExists = ifNotExists, tableOptions = tableOptions
```

Comment:
usually we would have a carriage return between args

Reply:
let me push the change with CR between args...